engine/src/state_chain_observer/sc_observer/mod.rs

```diff
@@ -525,6 +516,39 @@ where
     }.instrument(info_span!("SCObserver")).boxed()).await
 }
 
+/// Get the CFE events from the SC at the given block with version checking
+pub async fn get_cfe_events(
+	state_chain_client: Arc<impl StorageApi>,
+	block_info: BlockInfo,
+) -> Result<Vec<CfeEvent>, anyhow::Error> {
+	// First we check the version number of the CFE events
+	let cfe_events_version = state_chain_client
+		.storage_value::<pallet_cf_cfe_interface::CfeEventsVersion<Runtime>>(block_info.hash)
+		.await
+		.expect("Failed to get CFE events version");
```

Comment:
I think we should move away from using expects really, and given the code didn't previously, can we return the error?